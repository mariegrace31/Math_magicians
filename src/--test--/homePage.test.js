import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from '../components/HomePage';
import '@testing-library/jest-dom/extend-expect';


describe('Home page component', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Home />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Homepage component correctly', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    );
    const welcome = screen.queryByText('Welcome to our page!');
    expect(welcome).toBeInTheDocument();
  });
});