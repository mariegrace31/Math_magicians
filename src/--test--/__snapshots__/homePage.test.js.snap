// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component should match DOM Snapshot 1`] = `
<div>
  <nav>
    <h1
      className="Title"
    >
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <h3>
      Welcome to our page!
    </h3>
    <div
      className="paragraph"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Elementum eu facilisis sed odio. Ullamcorper eget nulla facilisi etiam. Risus pretium quam vulputate dignissim suspendisse. Mi bibendum neque egestas congue quisque egestas diam in. Quis auctor elit sed vulputate mi sit amet mauris. Quam quisque id diam vel quam elementum. Vitae ultricies leo integer malesuada nunc. Nam at lectus urna duis. Integer feugiat scelerisque varius morbi. Tortor posuere ac ut consequat semper viverra nam libero. Pretium vulputate sapien nec sagittis aliquam. Varius vel pharetra vel turpis nunc eget lorem dolor sed. Lobortis scelerisque fermentum dui faucibus in ornare quam viverra orci.Fames ac turpis egestas sed. Fames ac turpis egestas sed tempus. Sed vulputate odio ut enim blandit. Vitae et leo duis ut diam quam nulla. Ullamcorper sit amet risus nullam eget felis. Velit ut tortor pretium viverra suspendisse. Egestas fringilla phasellus faucibus scelerisque eleifend donec pretium.Ornare arcu odio ut sem nulla pharetra diam sit. Volutpat sed cras ornare arcu dui vivamus arcu. Ultricies lacus sed turpis tincidunt id aliquet risus feugiat.
      </p>
      <br />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Elementum eu facilisis sed odio. Ullamcorper eget nulla facilisi etiam. Risus pretium quam vulputate dignissim suspendisse. Mi bibendum neque egestas congue quisque egestas diam in. Quis auctor elit sed vulputate mi sit amet mauris. Quam quisque id diam vel quam elementum. Vitae ultricies leo integer malesuada nunc. Nam at lectus urna duis. Integer feugiat scelerisque varius morbi. Tortor posuere ac ut consequat semper viverra nam libero. Pretium vulputate sapien nec sagittis aliquam. Varius vel pharetra vel turpis nunc eget lorem dolor sed. Lobortis scelerisque fermentum dui faucibus in ornare quam viverra orci.Fames ac turpis egestas sed. Fames ac turpis egestas sed tempus. Sed vulputate odio ut enim blandit. Vitae et leo duis ut diam quam nulla. Ullamcorper sit amet risus nullam eget felis. Velit ut tortor pretium viverra suspendisse. Egestas fringilla phasellus faucibus scelerisque eleifend donec pretium.Ornare arcu odio ut sem nulla pharetra diam sit. Volutpat sed cras ornare arcu dui vivamus arcu. Ultricies lacus sed turpis tincidunt id aliquet risus feugiat.
      </p>
    </div>
  </main>
</div>
`;
